src/Chunk.h

```diff
@@ -107,11 +115,22 @@ class cChunk :
 	/** Marks all clients attached to this chunk as wanting this chunk. Also sets presence to cpQueued. */
 	void MarkRegenerating(void);
 
-	/** Returns true iff the chunk has changed since it was last saved. */
-	bool IsDirty(void) const {return m_IsDirty; }
+	/** Returns true if the chunk has changed since it was last saved.
+	If true, implies that CanUnload() is false. */
+	bool IsDirty(void) const {return m_SaveStatus != eSaveStatus::CLEAN; }
 
+	/** Returns true if the chunk is dirty, not saving, and not queued for save.
+	If true, implies that IsDirty() is true, and that CanUnload() is false. */
+	bool IsDirtyAndNotQueuedForSave(void) const {return m_SaveStatus == eSaveStatus::DIRTY; }
```

Comment:
The work of this comparison function should be done by the world storage thread.